src/query.rs

```diff
@@ -624,6 +626,7 @@ mod tests {
         crate_pair_name: &String,
         indexed_crate_new: &VersionedIndex<'_>,
         indexed_crate_old: &VersionedIndex<'_>,
+        parsed_query: Arc<IndexedQuery>,
```

Comment:
I'm not sure we need ownership here. We might be able to get away with just `&IndexedQuery` and avoid a bunch of cloning.